ts/features/fims/screens/FimsFlowHandlerScreen.tsx

```diff
@@ -32,44 +37,71 @@ export const FimsFlowHandlerScreen = (
 ) => {
   const { ctaUrl } = props.route.params;
   const dispatch = useIODispatch();
-  const errorState = useIOSelector(fimsErrorStateSelector);
+
+  useHeaderSecondLevel({ title: "", supportRequest: true });
 
   React.useEffect(() => {
     if (ctaUrl) {
       dispatch(fimsGetConsentsListAction.request({ ctaUrl }));
     }
   }, [ctaUrl, dispatch]);
 
+  const loadingState = useIOSelector(fimsLoadingStateSelector);
+  const consentsPot = useIOSelector(fimsConsentsDataSelector);
+  const errorState = useIOSelector(fimsErrorStateSelector);
+
+  if (errorState !== undefined) {
+    return (
+      <OperationResultScreenContent
+        pictogram="umbrellaNew"
+        title={errorState}
+        isHeaderVisible={true}
+      />
+    );
+  }
+  if (loadingState !== undefined) {
+    const subtitle =
+      loadingState === "in-app-browser" ? (
+        <View style={IOStyles.alignCenter}>
+          <LabelSmall color="grey-650" weight="Regular">
+            {I18n.t("FIMS.loadingScreen.in-app-browser.subtitle")}
+          </LabelSmall>
+        </View>
+      ) : (
+        <></>
+      );
+    const title = I18n.t(`FIMS.loadingScreen.${loadingState}.title`);
+
+    return (
+      <LoadingScreenContent contentTitle={title}>
+        {subtitle}
+      </LoadingScreenContent>
+    );
+  }
+
   return pipe(
-    errorState,
+    consentsPot,
+    pot.toOption,
     O.fold(
-      () => <FimsFlowSuccessBody />,
-      err => (
-        <OperationResultScreenContent
-          pictogram="umbrellaNew"
-          title={err.message}
-          isHeaderVisible={true}
-        />
-      )
+      () => <GenericErrorScreen />,
+      consents => <FimsFlowSuccessBody consents={consents} />
     )
   );
 };
 
-const FimsFlowSuccessBody = () => {
-  useHeaderSecondLevel({ title: "", supportRequest: true });
+const GenericErrorScreen = () => (
```

Comment:
What about using the `OperationResultsScreenContent` once instead of duplicating it for the error state? `title` can be an input parameter to this function